src/java.desktop/macosx/classes/sun/java2d/metal/MTLGraphicsConfig.java

```diff
@@ -145,9 +145,8 @@ public static MTLGraphicsConfig getConfig(CGraphicsDevice device,
         MTLRenderQueue rq = MTLRenderQueue.getInstance();
         rq.lock();
         try {
-            // getMTLConfigInfo() creates and destroys temporary
-            // surfaces/contexts, so we should first invalidate the current
-            // Java-level context and flush the queue...
+            // getMTLConfigInfo() creates new MTLContext, so we should first
+            // invalidate the current Java-level context and flush the queue...
```

Comment:
The old discussion was related not only to the comment but to the invalidateCurrentContext, do we need to do it?

Reply:
This is the only place where we use MTLContext.invalidateCurrentContext() - which when processed in MTLRenderQueue - clears some native stuff and nulls out both mtlc and dstOps pointers maintained in MTLRenderQueue.m. I think, this will be important once we get rid of SET_SCRATCH_SURFACE under JDK-8263309.

Reply:
But why you need to invalidate context here? Why do you need "clears some native stuff and nulls out both mtlc and dstOps pointers maintained in MTLRenderQueue.m"?

In OGL the getCGLConfigInfo() change the state of the OGL state due to "makeCurrentContext", this is why we need to update the lava level state to "invalid", otherwise we will get a mismatch between the state in the native and java state.

Reply:
I see that MTLContext.invalidateCurrentContext() is not needed here. Also, as this is the only place it is used, we can remove the method altogether.